lib/DataFetcher.dart

```diff
@@ -73,8 +76,46 @@ class DataFetcher {
     return new List();
   }
 
-  static Future<bool> uploadGrades(String filePath) {
-    return Future.value(true); // TODO send to server
+  static fetchFormData(FormClass.FormData data) async {
+    Dio dio = new Dio();
+
+    final String serverEndPoint = _api + "/user/profile";
+
+      FormData formData = new FormData.fromMap(({
+      "name": data.name,
+      "semester" : data.semester,
+      "school": data.school,
+      "reason": data.reason,
+      "study_time": data.studyTime,
+      "lectures": data.lectures,
+      "private": data.privateLessons,
+      "postgraduate": data.postGraduate,
+      "roomates": data.roommate,
+      "distance": data.distance,
+      "hobbies": data.hobbies,
+    }));
+
+    Response response = await dio.put(serverEndPoint, data: formData);
+
+    return response;
+
+  }
+
+  static uploadGrades(String filePath) async {
+    Dio dio = new Dio();
+
+    final String serverEndPoint = _api + "/grades/pdf";
```

Comment:
Remove the leading "/" as the _api variable has it already, which is causing the request to fail with a `404 Not found`.